### DISCUSSION TITLE
[2024-08-19] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#10845] [Weekly Demo Day]
* #10774 presented last time was merged.
* #10787 it was implemented but not working
* #10814

[Screencast from 2024-08-14 12-10-51.webm](https://github.com/user-attachments/assets/fcd5bfa4-b128-4a84-a19f-c14e78dae8c9)

* Now hurrying up with #10604 to finish it before my vacation.

**Comment by farmaazon** [comment]
@vitvakatu 
* The lion's share of last iteration was spent investigating #10527 
* #10592

**Comment by farmaazon** [comment]
@Frizi 
* #10735 - needed for cloud's ydoc server
* Fixed some follow issues: #10789 #10786 #10817
* #10803
* Bazel work is almost done for GUI components.

**Comment by kazcw** [comment]
- #10756

    - Better error detection in cases like: #10473

    - Complex lambda arguments:

```ruby
func = \a:Integer=1 (b:Integer = 2) ((c : Integer) = 3) -> a+b+c
```

- #10697

    - Better performance when editing large modules
    - Removes some potential pitfalls

- #10582 

https://github.com/user-attachments/assets/0d6bce44-735a-4deb-86c9-72a088ab1d59

- #10420

https://github.com/user-attachments/assets/1ad4d76a-1421-4d59-afb9-05f807877970

- #10800

https://github.com/user-attachments/assets/4d9b0d87-6d94-4053-8577-a2d4b6526f4c

- #10801

<img width="589" alt="Screenshot 2024-08-12 at 09 50 50" src="https://github.com/user-attachments/assets/bcd8860b-2755-414d-9833-0e95392df1b9">

- #10681 

- #10527

**Comment by JaroslavTulach** [comment]
Engine:
- work on native image for Ydoc - #10783
- emojis for node #10764
- caching self argument #10785
- dockerizing Ydoc #10811
- 10% speedup of startup - #10729
